posix/unix.c

```diff
@@ -674,6 +674,11 @@ static ssize_t recv(unsigned socket, void *buf, size_t len, int flags, struct so
 		for (;;) {
 			proc_lockSet(&s->lock);
 			if (s->type == SOCK_STREAM) {
+				if (flags & MSG_PEEK) {
+					err = _cbuffer_peek(&s->buffer, buf, len);
+					proc_lockClear(&s->lock);
+					break;
```

Comment:
Don't not break here - control data (`fdpass_unpack`) should be read too. Not sure if peek mode applies to control data, you should probably check how it works on Linux (see unix socket tests for sample code which send control data).

Reply:
It looks like under linux control data is peeked (not read) as well:
https://github.com/torvalds/linux/blob/master/net/unix/af_unix.c#L2620

Following our discussion, I've added a TODO for this

Reply:
It was above function declaration, possibly got hidden in the diff. Moved it inside `if`